reflex/app.py

```diff
@@ -269,6 +275,31 @@ def add_middleware(self, middleware: Middleware, index: Optional[int] = None):
         else:
             self.middleware.insert(index, middleware)
 
+    @staticmethod
+    def _generate_component(component: Component | ComponentCallable) -> Component:
```

Comment:
We can move this out of this class also, to a utils file

Reply:
I'll do it for the client side routing PR, since I also need to expand the 404 page over there before I pass it to wait_for_redir